core/src/main/java/io/seata/core/store/db/LockStoreDataBaseDAO.java

```diff
@@ -0,0 +1,356 @@
+        List<LockDO> locks = new ArrayList<>();
+        locks.add(lockDO);
+        return unLock(locks);
+    }
+
+    @Override
+    public boolean unLock(List<LockDO> lockDOs) {
+        Connection conn = null;
+        PreparedStatement ps = null;
+        try {
+            conn = logStoreDataSource.getConnection();
+            conn.setAutoCommit(true);
+
+            StringBuilder sb = new StringBuilder();
+            for(int i = 0; i< lockDOs.size(); i++){
+                sb.append("?");
+                if(i != (lockDOs.size() - 1)){
+                    sb.append(", ");
+                }
+            }
+            //batch release lock
+            String batchDeleteSQL = LockStoreSqls.getBatchDeleteLockSql(lockTable, sb.toString(), dbType);
+            ps = conn.prepareStatement(batchDeleteSQL);
+            ps.setString(1, lockDOs.get(0).getXid());
+            for(int i = 0; i< lockDOs.size(); i++){
+                ps.setString(i+2, lockDOs.get(i).getRowKey());
+            }
+            return ps.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new StoreException(e);
+        } finally {
+            if(ps != null){
+                try {
+                    ps.close();
+                } catch (SQLException e) {
+                }
+            }
+            if (conn != null) {
+                try {
+                    conn.close();
+                } catch (SQLException e) {
+                }
+            }
+        }
+    }
+
+    @Override
+    public boolean isLockable(List<LockDO> lockDOs) {
+        Connection conn = null;
+        try {
+            conn = logStoreDataSource.getConnection();
+            conn.setAutoCommit(true);
+            if(!checkLockable(conn, lockDOs)){
+                return false;
+            }
+            return true;
+        } catch (SQLException e) {
+            throw new StoreException(e);
+        } finally {
+            if (conn != null) {
+                try {
+                    conn.close();
+                } catch (SQLException e) {
+                }
+            }
+        }
+    }
+
+    /**
+     * Do acquire lock boolean.
+     *
+     * @param conn   the conn
+     * @param lockDO the lock do
+     * @return the boolean
+     */
+    protected boolean doAcquireLock(Connection conn, LockDO lockDO) {
+        PreparedStatement ps = null;
+        ResultSet rs = null;
+        try {
+            //insert
+            String insertLockSQL = LockStoreSqls.getInsertLockSQL(lockTable, dbType);
+            ps = conn.prepareStatement(insertLockSQL);
+            ps.setString(1, lockDO.getXid());
+            ps.setLong(2, lockDO.getTransactionId());
+            ps.setLong(3, lockDO.getBranchId());
+            ps.setString(4, lockDO.getResourceId());
+            ps.setString(5, lockDO.getTableName());
+            ps.setString(6, lockDO.getPk());
+            ps.setString(7, lockDO.getRowKey());
+            return ps.executeUpdate() > 0;
```

Comment:
Throw DuplicateKeyException is equal to 'return false', which mean acquireLock failed both.

Reply:
> Throw DuplicateKeyException is equal to 'return false', which mean acquireLock failed both.

ok，i see，if DuplicateKeyException is attributed to a failure to acquire locks, rollback is  needed to consider，by the way ，how about batch insert which need not to consider rollback？

Reply:
> how it happened when ps.executeUpdate() <= 0 ? i think if there is a row exists,it will throw DuplicateKeyException ?

I will optimize it, but may not be in this pr.

Reply:
I will optimize it, but may not be in this pr.